src/integration/java/org/springframework/data/couchbase/repository/SimpleCouchbaseRepositoryTests.java

```diff
@@ -220,6 +238,66 @@ public void shouldTakeVersionIntoAccountWhenDoingMultipleUpdates() {
     }
   }
 
+  @Test
+  public void shouldUpdateDocumentConcurrently() throws Exception {
+    final String key = testName.getMethodName();
+    remove(key);
+
+    final AtomicLong counter = new AtomicLong();
+    VersionedData initial = new VersionedData(key, "value-initial");
+    versionedDataRepository.save(initial);
+    assertNotEquals(0L, initial.version);
+
+    Callable<Void> task = new Callable<Void>() {
+      @Override
+      public Void call() throws Exception {
+        boolean saved = false;
+        while(!saved) {
+          long counterValue = counter.incrementAndGet();
+          VersionedData messageData = versionedDataRepository.findOne(key);
+          messageData.data = "value-" + counterValue;
+          try {
+            versionedDataRepository.save(messageData);
+            saved = true;
+          } catch (OptimisticLockingFailureException e) {
+          }
```

Comment:
maybe add a counter here too, to assert that exactly 1 out of 5 concurrent updates succeeded?